UM/Trust.py

```diff
@@ -451,9 +461,29 @@ def signedFolderPreStorageCheck(self, path: str) -> bool:
                     self._violation_handler(f"Manifest '{manifest_path}' is not properly self-signed in '{path}'.")
                     return False
 
-                # Otherwise, as far as this quick pre check is concerned, there is nothing wrong:
-                Logger.log("i", f"Central storage file signed correctly for '{path}'.")
-                return True
+            # Check if the central storage file doesn't contain files that would be moved outside the plugin folder,
+            #   or files that would be moved to outside of the central storage location:
+            with open(central_storage_filename, "r", encoding = "utf-8") as central_storage_file:
+                central_storage_list = json.loads(central_storage_file.read())
```

Comment:
The json could be malformatted. Not sure if we should catch that (and log it) or just accept that the exception bubbles up into a critical error.

Reply:
It is caught, see the try-statement on line 432.